### PR TITLE
[IM] Reject multiple commands in a same invoke transaction on the server side

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem

The protocol supports putting multiple commands in the same invoke request, but this is a feature for 1.x. We did not reject multiple commands on the server side for now.

#### Change overview

Add a check before handling the commands.

#### Testing
- Adds TestCommandHandlerRejectMultipleCommands

### CODE REVIEW

File: src/app/CommandHandler.cpp [hunk 1]
```diff
@@ -125,6 +126,15 @@ CHIP_ERROR CommandHandler::ProcessInvokeRequest(System::PacketBufferHandle && pa
     }
 
     invokeRequests.GetReader(&invokeRequestsReader);
+
+    {
+        // We don't support handling multiple commands but the protocol is ready to support it in the future, reject all of them and
+        // IM Engine will send a status response.
+        size_t commandCount = 0;
+        TLV::Utilities::Count(invokeRequestsReader, commandCount, false /* recurse */);
+        VerifyOrReturnError(commandCount == 1, CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
```
Comment (bzbarsky-apple) [inline src/app/CommandHandler.cpp:135]: What is the observable behavior here on the client side?

Comment (erjiaqing) [reply src/app/CommandHandler.cpp:135]: It will reject to encode such payload (CHIP_ERROR_INCORRECT_STATE)

Comment (bzbarsky-apple) [reply src/app/CommandHandler.cpp:135]: No, I mean if a client that is not the SDK sends multiple commands in a single invoke, what behavior will it observe?  Will it get back an error StatusResponse (if so what kind)?  Will it get back nothing?  Something else?

Comment (erjiaqing) [reply src/app/CommandHandler.cpp:135]: This is what the TestCommandHandlerRejectMultipleCommands covers, it will get a StatusResponse.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by cjandhyala** [comment]
@erjiaqing can we have spec update as well so that we can update our test cases accordingly.

**Comment by bzbarsky-apple** [comment]
@cjandhyala Per spec, multiple commands in an invoke are allowed.  It's just not supported in the SDK for 1.0, just like a bunch of other things the spec allows that we don't support yet.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**